s2stac/src/main/scala/crawler/Crawler.scala

```diff
@@ -0,0 +1,484 @@
+    if (history.years.contains(yearHistory)) {
+      printInfo[F](s"Year ${yearHistory.toID} already exists") *>
+        Sync[F].pure(CrawlerState.initial)
+    } else {
+      val yearCatalog = StacCatalog(
+        "1.0.0-beta1",
+        Nil,
+        s"catalog-year-${yearHistory.toID}",
+        None,
+        s"Catalog holding descendants of year ${yearHistory.year}",
+        List(
+          StacLink("../catalog.json", StacLinkType.Parent, Some(`application/json`), None),
+          StacLink("../../../../collection.json", StacLinkType.StacRoot, Some(`application/json`), None)
+        )
+      )
+      printInfo[F](s"Year ${yearHistory.toID} did not exist!") map { _ =>
+        CrawlerState.yearHistory(yearHistory, yearCatalog) `combine` CrawlerState.squareCatalogChild(
+          SquareCatalogId(yearHistory.square.toID),
+          yearHistory
+        )
+      }
+    }
+
+  private def makeMonthParent(history: CrawlerState, monthHistory: History.MonthHistory) =
+    if (history.months.contains(monthHistory)) {
+      printInfo[F](s"month ${monthHistory.toID} already exists") *>
+        Sync[F].pure(CrawlerState.initial)
+    } else {
+      val monthCatalog = StacCatalog(
+        "1.0.0-beta1",
+        Nil,
+        s"catalog-month-${monthHistory.toID}",
+        None,
+        s"Catalog holding descendants of month ${monthHistory.month}",
+        List(
+          StacLink("../catalog.json", StacLinkType.Parent, Some(`application/json`), None),
+          StacLink("../../../../../collection.json", StacLinkType.StacRoot, Some(`application/json`), None)
+        )
+      )
+      printInfo[F](s"month ${monthHistory.toID} did not exist!") map { _ =>
+        CrawlerState.monthHistory(monthHistory, monthCatalog) `combine` CrawlerState.yearCatalogChild(
+          YearCatalogId(monthHistory.year.toID),
+          monthHistory
+        )
+      }
+    }
+
+  private def makeDayParent(history: CrawlerState, dayHistory: History.DayHistory) =
+    if (history.days.contains(dayHistory)) {
+      printInfo[F](s"day ${dayHistory.toID} already exists") *>
+        Sync[F].pure(CrawlerState.initial)
+    } else {
+      val dayCatalog = StacCatalog(
+        "1.0.0-beta1",
+        Nil,
+        s"catalog-day-${dayHistory.toID}",
+        None,
+        s"Catalog holding descendants of day ${dayHistory.day}",
+        List(
+          StacLink("../catalog.json", StacLinkType.Parent, Some(`application/json`), None),
+          StacLink("../../../../../../collection.json", StacLinkType.StacRoot, Some(`application/json`), None)
+        )
+      )
+      printInfo[F](s"day ${dayHistory.toID} did not exist!") map { _ =>
+        CrawlerState.dayHistory(dayHistory, dayCatalog) `combine` CrawlerState.monthCatalogChild(
+          MonthCatalogId(dayHistory.month.toID),
+          dayHistory
+        )
+      }
+    }
+
+  private def getItemAssets(tileInfo: L1CTileInfo): Map[String, StacItemAsset] = {
+    Map(
```

Comment:
There's a few things missing here that are in the published Sentinel 2 STAC that I think we should consider adding:
 - tile info json - I think this makes sense as a metadata file to include
 - thumbnail - which looks like it's a file called "preview.jpg" in the directory of the rest of the data
 - metadata xml - also in the same directory as the rest of the data as "metadata.xml"
 - true color image - also in the same directory as the rest of the data as "TKI.jp2"

I agree we don't want to go overboard - we'll need some input from the R&D team about what the useful assets are, but parity with the extant Sentinel 2 STAC datasets.

Reply:
b1bd9bb